Civi/RemoteParticipant/Event/ChangingEvent.php

```diff
@@ -65,6 +74,23 @@ public abstract function getParticipant();
      */
     public abstract function &getParticipantData();
 
+    /**
+     * @throws \Exception
+     *    if another contact ID has already been set
+     */
+    public function setContactID($contact_id): void
+    {
+        $contact_id = (int) $contact_id;
+        if (0 !== $contact_id) {
+            if ($this->getContactID() && $this->getContactID() !== $contact_id) {
+                throw new \Exception(
+                  'Conflicting contact IDs, there is a programming error'
+                );
+            }
```

Comment:
What scenario is this validating?

Reply:
That's code that already existed that was moved from here https://github.com/systopia/de.systopia.remoteevent/blob/65b66354dc8f3c023f8c27071e9cfc538332040e/Civi/RemoteParticipant/Event/RegistrationEvent.php#L210 into the super class. I currently cannot say why it exists.